mapreader/download/downloader.py

```diff
@@ -97,12 +101,45 @@ def _download_map(self, grid_bb: GridBoundingBox, map_name: str | None) -> bool:
             The grid bounding box of the map
         map_name : str, optional
             Name to use when saving the map, by default None
+        force : bool, optional
+            Whether to force download, by default False
 
         Returns
         -------
         bool
             True if map was successfully downloaded, False if not.
         """
+        try:
+            # get url for single tile to estimate size
+            tile_url = self.downloader.generate_tile_url(grid_bb.lower_corner, 0)
+
+            user_agent = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/35.0.1916.47 Safari/537.36"
+            headers = {"User-Agent": user_agent}
+            request = urllib.request.Request(tile_url, None, headers)
+            response = urllib.request.urlopen(request)
+            # get size of single tile
+            size_bytes = len(response.read())
+            size_mb = size_bytes * 1e-6
+
+            # get total number of tiles
+            no_tiles = grid_bb.covered_cells
+            total_size_mb = no_tiles * size_mb
+
+            if total_size_mb > 100:
+                print(f"[WARNING] Estimated total size: {total_size_mb * 1e-3:.2f}GB.")
+                if not force:
+                    raise ValueError(
+                        "[ERROR] This will download a lot of data. Please confirm download by setting ``force=True``."
```

Comment:
Include the actual amount in the message. Ask to confirm, don't phrase it as if something has gone wronf